config/v1/types_infrastructure.go

```diff
@@ -657,6 +702,23 @@ type OvirtPlatformStatus struct {
 
 	// deprecated: as of 4.6, this field is no longer set or honored.  It will be removed in a future release.
 	NodeDNSIP string `json:"nodeDNSIP,omitempty"`
+
+	// loadBalancerType defines if a user-managed or openshift-managed loadbalancer
+	// is to be used for the OpenShift API and Ingress endpoints.
+	// When set to OpenShiftManaged the static pods in charge of API and Ingress traffic load-balancing
+	// defined in the machine config operator will be deployed.
+	// When set to UserManaged these static pods will not be deployed and it is expected that
+	// the load balancer is configured out of band by the deployer.
+	// When omitted, this means no opinion and the platform is left to choose a reasonable default.
+	// The default value is OpenShiftManaged.
+	// +default="OpenShiftManaged"
+	// +kubebuilder:default:="OpenShiftManaged"
+	// +kubebuilder:validation:Enum:="OpenShiftManaged";"UserManaged"
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:XValidation:rule="oldSelf == '' || self == oldSelf",message="loadBalancerType is immutable once set"
+	// +openshift:enable:FeatureSets=TechPreviewNoUpgrade
+	// +optional
+	LoadBalancerType PlatformLoadBalancerType `json:"loadBalancerType,omitempty"`
```

Comment:
Is there any way to DRY this?

Reply:
if by drying it you mean make it shorter, no. I was asked to put these validations for each of the platforms, so we can change them individually and not globally.

Reply:
`D`on't `R`epeat `Y`ourself.

The answer might be know, but this is a moderate complexity block repeated several time. The probability of drift through maintenance is high. The answer might be no, but it's worth considering what we could do to mitigate it.

Reply:
I know and initially I put the validations at the Type level, but I was told to not do that and repeat it by platform.

Reply:
The answer is no 😉 Yes, I'd love to see avoiding the duplication, but we've seen issues in the past with sharing types and their validations so it's better to put them explicitly here.

For example, if this was shared, one platform cannot evolve their API without evolving the rest of the platforms, generally, that's a negative. Imagine for example if a certain platform wants to support an additional enum value, why do the others need to support this?